pkg/gardenlet/controller/backupentry/add.go

```diff
@@ -138,3 +139,24 @@ func getBackupBucketLastOperation(obj client.Object) *gardencorev1beta1.LastOper
 
 	return backupBucket.Status.LastOperation
 }
+
+// restoreAfterMigratePredicate returns a predicate which returns false if the core.gardener.cloud/v1beta1.BackupEntry has been
+// successfully migrated and does not have the `gardener.cloud/operation: restore` annotation.
+// It returns true in all other cases.
+func restoreAfterMigratePredicate() func(obj client.Object) bool {
+	return func(obj client.Object) bool {
+		backupEntry, ok := obj.(*gardencorev1beta1.BackupEntry)
+		if !ok {
+			return false
+		}
+
+		isMigrateSucceeded := backupEntry.Status.LastOperation != nil && backupEntry.Status.LastOperation.State == gardencorev1beta1.LastOperationStateSucceeded && backupEntry.Status.LastOperation.Type == gardencorev1beta1.LastOperationTypeMigrate
+		hasRestoreAnnotation := backupEntry.GetAnnotations()[v1beta1constants.GardenerOperation] == v1beta1constants.GardenerOperationRestore
+
+		if !isMigrateSucceeded || hasRestoreAnnotation {
+			return true
+		}
+
+		return false
```

Comment:
Some simplifications and readability
```suggestion
	isMigrateSucceeded := backupEntry.Status.LastOperation != nil &&
		backupEntry.Status.LastOperation.State == gardencorev1beta1.LastOperationStateSucceeded &&
		backupEntry.Status.LastOperation.Type == gardencorev1beta1.LastOperationTypeMigrate

	hasRestoreAnnotation := backupEntry.GetAnnotations()[v1beta1constants.GardenerOperation] == v1beta1constants.GardenerOperationRestore

	return !isMigrateSucceeded || hasRestoreAnnotation
```